pandas_gbq/schema/bigquery_to_pyarrow.py

```diff
@@ -0,0 +1,275 @@
+# Copyright (c) 2025 pandas-gbq Authors All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+from typing import Any, Union
+import warnings
+
+import db_dtypes
+from google.cloud import bigquery
+import pyarrow
+
+
+def pyarrow_datetime():
+    return pyarrow.timestamp("us", tz=None)
+
+
+def pyarrow_numeric():
+    return pyarrow.decimal128(38, 9)
+
+
+def pyarrow_bignumeric():
+    # 77th digit is partial.
+    # https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#decimal_types
+    return pyarrow.decimal256(76, 38)
+
+
+def pyarrow_time():
+    return pyarrow.time64("us")
+
+
+def pyarrow_timestamp():
+    return pyarrow.timestamp("us", tz="UTC")
+
+
+# Prefer JSON type built-in to pyarrow (adding in 19.0.0), if available.
+# Otherwise, fallback to db-dtypes, where the JSONArrowType was added in 1.4.0,
+# but since they might have an older db-dtypes, have string as a fallback for that.
+if hasattr(pyarrow, "json_"):
+    json_arrow_type = pyarrow.json_(pyarrow.string())
+elif hasattr(db_dtypes, "JSONArrowType"):
+    json_arrow_type = db_dtypes.JSONArrowType()
+else:
+    json_arrow_type = pyarrow.string()
```

Comment:
I had a change of heart in https://github.com/googleapis/python-bigquery/pull/1876 For `to_arrow()`, we should emulate the BQ Storage Read API as closely as possible.

For `read_gbq()`, that's where I'd like to use the extension type(s) if available.